src/models/TaskList.js

```diff
@@ -37,4 +52,26 @@ export class TaskList {
         }
         console.log("num of tasks of deleting: " + this.tasks.length);
     }
+
+    // Static method to serialize a TaskList object into a list of dictionaries
+    static serialize(taskList) {
+        const taskDictionaries = [];
+        for (const task of taskList.tasks) {
+            taskDictionaries.push({
+                name: task.name,
```

Comment:
You can probably use your task serializer here... this is why it was created.

`Task.serialize(task)`

At this point, might just be worth forgetting about this, but remember why you made the functions you made.